### PR TITLE
Test Hiccup/Jitter Monitor

### PR DESCRIPTION
**Pull request by jerrinot**
The current impl starts a monitoring thread and never stop it. I am not sure if that's a problem as I assume tests are executed in its own JVM anyway?

### CODE REVIEW

**Comment by mdogan** [comment]
Good and helpful work! 
I haven't really look at `JitterRecorder` impl.  I'm assuming you know what you do there :)

Do you know is it possible to attach junit rule to test runner? Currently only tests extending `HazelcastTestSupport` are measured.

**Comment by jerrinot** [comment]
@mdogan it is actually implemented as a rule. the `HazelcastTestSupport` just [adds this](https://github.com/hazelcast/hazelcast/pull/9482/files#diff-1f0f61976faa2b7e01b0bb185336b895R92) rule by default. However nothing is preventing you to use the rule in arbitrary tests.

**Comment by mdogan** [comment]
@jerrinot: I see, thanks. 

So what's preventing us from merging this?

**Comment by jerrinot** [comment]
Currently it starts a [monitoring thread](https://github.com/hazelcast/hazelcast/pull/9482/files#diff-484693c50114046bbb9f978b3aa0b473R57) and never stops it. 

I don't know if that's a problem. I assume tests are always executed in a new JVM, but I am not 100% sure.

**Comment by mdogan** [comment]
What if we make jitter-rule disabled by default and enable it via system property? We can enable it on jenkins always.

**Comment by jerrinot** [comment]
I added Jenkins auto-detection + a property for explicit control. let's see whether it works at all.

**Comment by hasancelik** [comment]
run-lab-run

**Comment by jerrinot** [comment]
previous failures were intentional - to test reporting on a test failure.

File: hazelcast/src/test/java/com/hazelcast/test/jitter/JitterRule.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        }
+
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                long startTime = System.currentTimeMillis();
+                try {
+                    base.evaluate();
+                } catch (Throwable t) {
+                    long endTime = System.currentTimeMillis();
```
Comment (mmedenjak) [inline hazelcast/src/test/java/com/hazelcast/test/jitter/JitterRule.java]: We should possibly log the jitter no matter if the test failed or succeed. A rule that is executed after the JitterRule can cause the test to fail, depending on the order in which rules are applied.

**Pull request merged**